Remove documentation for build retention (#2024)
buildkite/docs: pages/pipelines/build_export.md

> # Configuring build export

Reviewer: If you update the nav, also update this heading.

Each [Buildkite plan](https://buildkite.com/pricing) has a maximum build retention

> period, after which your oldest builds will be automatically deleted.

Reviewer: tightening slightly while we're here:

```suggestion
period, after which your oldest builds are automatically deleted.
```
